### PR TITLE
Add explicit example for RAG chat templating

### PR DESCRIPTION
**Pull request by A-Duss**
# What does this PR do?

As discussed with @Rocketknight1 in #33421, this commit updates the `chat_templating` documentation to:
- Clarify that the documents input is not yet supported by most models.
- Provide an example with a model that do support it.
- Offer guidance on how to check if a model supports the `documents` feature.

Hoping to improve readers understanding of the feature.

Might be beneficial to update the Spanish #28936, Japanese #28301 , Korean #20179 , and Chinese #26803 corresponding docs, I did not risk myself as I do not speak those languages fluently enough. es, ja, ko, zh

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. 
@stevhliu @Rocketknight1

### CODE REVIEW

File: docs/source/en/chat_templating.md [hunk 1]
```diff
@@ -616,22 +616,63 @@ than the JSON schemas used for tools, no helper functions are necessary.
...
+print(gen_text)
 ```
 
+**Note on `documents` input:**
+
+The `documents` input is not widely supported by many models, and chat templates often discard this input. Users should be cautious about relying on `documents` unless the model explicitly supports it.
+
+To verify if a model supports the `documents` input, check if it is included in the `chat_template` key within the `tokenizer_config.json` file in the model's card.
+
+As an example, Cohere's [Command-R](https://huggingface.co/CohereForAI/c4ai-command-r-08-2024) and [Command-R+](https://huggingface.co/CohereForAI/c4ai-command-r-plus-08-2024) support the `documents` input through their `rag` chat template. You can see additional examples of grounded generation using this feature in their model cards.
```
Comment (Rocketknight1) [inline docs/source/en/chat_templating.md:672]:
```suggestion
<Tip>

The `documents` input for retrieval-augmented generation is not widely supported, and many models have chat templates which simply ignore this input.

To verify if a model supports the `documents` input, you can read its model card, or `print(tokenizer.chat_template)` to see if the `documents` key is used anywhere.

One model class that does support it, though, is Cohere's [Command-R](https://huggingface.co/CohereForAI/c4ai-command-r-08-2024) and [Command-R+](https://huggingface.co/CohereForAI/c4ai-command-r-plus-08-2024), through their `rag` chat template. You can see additional examples of grounded generation using this feature in their model cards.

</Tip>
```

**Review by Rocketknight1 — APPROVED** [review]
This looks good! I ran the example successfully, and Command-R reported that "interestingly enough", man has always dreamed of destroying the moon. Definitely feel like it was wryly aware that the sources we'd given it might not have been entirely accurate, lol.

I made a suggestion to replace the section you added at the end with a `Tip` box, but other than that this should be ready to merge soon.

**Review by stevhliu — APPROVED** [review]
LGTM, thanks!

**Comment by A-Duss** [comment]
Thanks @Rocketknight1 for the tweaking, I committed your suggestion. 😄 
Should be ready to merge now. @stevhliu ? (not sure if I am supposed to @ you now, sorry if I am not)

**Comment by Rocketknight1** [comment]
Yes, it's ready to merge, and don't worry, I can approve it, since it's just a doc change.

Thank you, and congratulations on your first PR!

**Pull request merged**